### PR TITLE
Initial docs and API proposal

### PR DESCRIPTION
**Pull request by theogf**
This is the initial PR to have an idea of the API
I will not write too much here as most of the content of this PR  is described in the docs: https://juliagaussianprocesses.github.io/AugmentedGPLikelihoods.jl/previews/PR1/

This introduce only one likelihood (Bernoulli) with an associated tutorial to give a clear idea of how this would work.

### CODE REVIEW

File: docs/src/index.md [hunk 1]
```diff
@@ -6,9 +6,125 @@ CurrentModule = AugmentedGPLikelihoods
...
+```@docs
+init_aux_variables
+aux_sample!
+aux_sample
+sample_shift
+sample_rate
+```
+First [`init_aux_variables`](@ref) properly initializes a `NamedTuple` of `Vector`(s) of
+auxiliary variables to be modified in place during sampling.
+[`aux_sample!`](@ref) will sample in-place from the full conditional $$p(\Omega|y,f)$$ and return another `NamedTuple`, for no-inplace operation see [`aux_sample`].
```
Comment (st--) [inline docs/src/index.md:75]: if it's in-place, shouldn't it return the same tuple? or what am I missing?

File: src/api.jl [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+Return the shift of the first variational natural parameter
+$\eta_1=\Sigma^{-1}\mu$ given the auxiliary variables in `\Omega`
+and the observations `y`
+"""
+sample_shift
+
+@doc raw"""
+    sample_rate(lik::Likelihood, Ω, y) -> Tuple
+
+Return two times the shift of the second natural parameter
```
Comment (st--) [inline src/api.jl:70]: I find it quite confusing that `shift` is shift_1 and `rate` is 2*shift_2 (also for vi_ below)...

Comment (theogf) [reply src/api.jl:70]: I fully agree but this allows the simpler updates for sigma = inv(inv(K) + rate),

File: docs/src/index.md (hunk 1 above)
Comment (theogf) [reply docs/src/index.md:75]: You're right it's just bad phrasing!

File: src/api.jl [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+"""
+aux_sample!
+
+"""
+    aux_sample([rng::AbstractRNG], lik::Likelihood, y, f) -> NamedTuple
+
+Sample the auxiliary variables `Ω` in a new `NamedTuple`.
+For the explanation see [`aux_sample!`].
+"""
+aux_sample
```
Comment (st--) [inline src/api.jl:37]:
```suggestion
aux_update_delta_eta1
```

File: src/api.jl [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+    aux_posterior(lik::Likelihood, y, q_f) -> NamedTuple
+
+Compute the optimal posterior of the auxiliary variables in a new
+`NamedTuple`.
+See also [`aux_posterior!`](@ref)
+"""
+aux_posterior
+
+@doc raw"""
+    sample_shift(lik::Likelihood, Ω, y) -> Tuple
```
Comment (st--) [inline src/api.jl:59]:
```suggestion
    aux_update_delta_eta1(lik::Likelihood, Ω, y) -> Tuple
```

File: src/api.jl [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+vi_shift
+
+
+@doc raw"""
+    vi_rate(lik::Likelihood, Ω, y) -> Tuple
+
+Return two times the shift of the second natural parameter $\eta_2=-\frac{1}{2}\Sigma^{-1}$
+given the auxiliary posteriors contained in `\Omega` and the observations `y`
+"""
+vi_rate
```
Comment (st--) [inline src/api.jl:91]:
```suggestion
aux_update_expected_delta_eta2
```

**Pull request merged**